src/components/Navbar/GNB.tsx

```diff
@@ -1,37 +1,47 @@
-import SmallMainLogo from "assets/icon/main_logo_small.svg";
-import MainLogo from "assets/icon/main_logo.svg";
-import { useLocation } from "react-router-dom";
-import { Button } from "components/Button";
-import useDeviceState from "hooks/useDeviceState";
-import { NavLink } from "react-router-dom";
-import * as S from "./Navbar.style";
-
-export default function GNB() {
-  const { pathname } = useLocation();
-  const { isMobileWidth } = useDeviceState();
-
-  return (
-    <S.NavbarContainer>
-      <NavLink to="/">
-        <img src={isMobileWidth ? SmallMainLogo : MainLogo} alt="main-logo" />
-      </NavLink>
-      <S.NavbarLinkContainer>
-        <NavLink
-          to="/free-board"
-          style={({ isActive }) => ({ color: isActive ? "var(--blue)" : "" })}
-        >
-          자유게시판
-        </NavLink>
-        <NavLink
-          to="/items"
-          style={({ isActive }) => ({
-            color: isActive || pathname === "/additem" ? "var(--blue)" : "",
-          })}
-        >
-          중고마켓
-        </NavLink>
-      </S.NavbarLinkContainer>
-      <Button.Link to="/login">로그인</Button.Link>
-    </S.NavbarContainer>
-  );
-}
+import { NavLink, useLocation } from "react-router-dom";
+import { useAtomValue } from "jotai";
+import Button from "components/Button";
+import useDeviceState from "hooks/useDeviceState";
+import { loginState } from "contexts/atoms/users";
+import SmallMainLogo from "assets/icon/main_logo_small.svg";
+import MainLogo from "assets/icon/main_logo.svg";
+import profile from "assets/icon/profile.svg";
+import COLORS from "styles/palette";
+import * as S from "./Navbar.style";
+
+export default function GNB() {
+  const { pathname } = useLocation();
+  const { isMobileWidth } = useDeviceState();
+  const isLogin = useAtomValue(loginState);
+
+  return (
+    <S.NavbarContainer>
+      <NavLink to="/">
+        <img src={isMobileWidth ? SmallMainLogo : MainLogo} alt="main-logo" />
+      </NavLink>
+      <S.NavbarLinkContainer>
+        <NavLink
+          to="/free-board"
+          style={({ isActive }) => ({
+            color: isActive ? `${COLORS.BLUE}` : "",
+          })}
+        >
```

Comment:
styled components로 해당 컴포넌트를 확장해서 사용하는건 어떨까요?
style props에 복잡한 UI 분기처리를 숨겨줄수 있을것같아요 :) 

```suggestion
        <StyledNavLink
          to="/free-board"
        >
```